src/python/pants/backend/python/rules/prepare_chrooted_python_sources_test.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pathlib import PurePath
+from typing import List, Optional
+from unittest.mock import Mock
+
+from pants.backend.python.rules.prepare_chrooted_python_sources import (
+  ChrootedPythonSources,
+  prepare_chrooted_python_sources,
+)
+from pants.build_graph.address import Address
+from pants.build_graph.files import Files
+from pants.engine.legacy.graph import HydratedTarget, HydratedTargets
+from pants.engine.rules import RootRule
+from pants.engine.selectors import Params
+from pants.rules.core import strip_source_roots
+from pants.testutil.option.util import create_options_bootstrapper
+from pants.testutil.test_base import TestBase
+
+
+class PrepareChrootedPythonSourcesTest(TestBase):
+
+  @classmethod
+  def rules(cls):
+    return (
+      *super().rules(),
+      *strip_source_roots.rules(),
+      prepare_chrooted_python_sources,
+      RootRule(HydratedTargets),
+    )
+
+  def make_hydrated_target(
+    self, *, source_paths: List[str], type_alias: Optional[str] = None,
+  ) -> HydratedTarget:
+    adaptor = Mock()
+    adaptor.type_alias = type_alias
+    adaptor.sources = Mock()
+    adaptor.sources.snapshot = self.make_snapshot({fp: "" for fp in source_paths})
+    address = Address(spec_path=PurePath(source_paths[0]).parent.as_posix(), target_name="target")
+    return HydratedTarget(address=address, adaptor=adaptor, dependencies=())
+
+  def test_adds_missing_inits_and_strips_source_roots(self) -> None:
```

Comment:
I didn't stress test all the different combinations of source root stripping because that is already tested in `strip_source_roots_test.py`.

Let me know if you can think of any further edge cases with injecting `__init__.py`, though. One thing we could do is to ensure that a pre-existing file still keeps its original content. It would make the test setup a bit more complex, but is probably worth it? Thoughts?